server/models/FiltersModel/index.js

```diff
@@ -0,0 +1,43 @@
+import { STUDIES_TO_OMIT, TRUE_STRING, FALSE_STRING } from '../../constants'
+
+const FiltersModel = {
+  calculateFilters: function (filters, userAccess) {
+    if (!filters) {
+      this.defaultFilters.sites = userAccess.filter(
```

Comment:
I think we want to create new objects that we return from each call instead of assigning attributes to an existing object. So something like this instead:

```js
import { STUDIES_TO_OMIT, TRUE_STRING, FALSE_STRING } from '../../constants'

const DEFAULT_FILTERS = {
    chrcrit_part: [
      { name: 'HC', value: TRUE_STRING },
      { name: 'CHR', value: TRUE_STRING },
      { name: 'Missing', value: TRUE_STRING },
    ],
    included_excluded: [
      { name: 'Included', value: TRUE_STRING },
      { name: 'Excluded', value: FALSE_STRING },
      { name: 'Missing', value: FALSE_STRING },
    ],
    sex_at_birth: [
      { name: 'Male', value: TRUE_STRING },
      { name: 'Female', value: TRUE_STRING },
      { name: 'Missing', value: TRUE_STRING },
    ],
    sites: [],
  }

const FiltersModel = {
  calculateFilters: function (filters, userAccess) {
    if (!filters) {
      return {
        ...DEFAULT_FILTERS,
        sites: userAccess.filter((site) => !STUDIES_TO_OMIT.includes(site)),
      }
    } else {
      return {
        ...filters,
        sites: Object.values(filters.sites).map(({ value }) => value),
      }
    }
  },
  normalizeFilters: (filters) => {
    return {
      ...filters,
      sites: filters.sites.map((site) => ({ label: site, value: site })),
    }
  },
}

export default FiltersModel
```

Reply:
we have a similar pattern on this model 
https://github.com/AMP-SCZ/dpdash/blob/3b9e0dca38624f964806bdb6060eac7e0e412697/server/models/UserModel/index.js#L58-L79

Reply:
yeah except that is creating a new object with each call

Reply:
👍 [d95c571](https://github.com/AMP-SCZ/dpdash/pull/441/commits/d95c571be06267f0f7e05e8433d13db1177af14c)